truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java

```diff
@@ -2615,159 +2618,143 @@ public CallTarget getCallTarget() {
         }
     }
 
-    @CoreMethod(names = "pack", required = 1)
+    @CoreMethod(names = "pack", required = 1, taintFromParameter = 0)
     public abstract static class PackNode extends ArrayCoreMethodNode {
 
-        @Child private CallDispatchHeadNode toStringNode;
+        @Child private TaintNode taintNode;
 
         public PackNode(RubyContext context, SourceSection sourceSection) {
             super(context, sourceSection);
         }
 
-        // TODO CS 3-Mar-15 to be honest these two specialisations are a bit sneaky - we'll get rid of them ASAP
+        @Specialization(guards = "byteListsEqual(format, cachedFormat)")
+        public RubyString packCached(
+                VirtualFrame frame,
+                RubyArray array,
+                RubyString format,
+                @Cached("privatizeByteList(format)") ByteList cachedFormat,
+                @Cached("create(compileFormat(format))") DirectCallNode callPackNode) {
```

Comment:
The DSL expands it into `DirectCallNode.create(compileFormat(format))`. There is a lookup order for where it finds methods in guards, and one place is static members of the parameters's class. It's not just copied and pasted into the DSL generated code - there is a specific lookup.